spec/expression.dd

```diff
@@ -96,10 +96,38 @@ $(H4 Assignment Operator Expressions)
     $(P except that:)
 
     $(UL
-        $(LI operand $(D a) is only evaluated once)
+        $(LI operand $(D a) is only evaluated once, see below.)
         $(LI overloading $(I op) uses a different function than overloading $(I op)= does)
         $(LI the left operand of $(D >>>=) does not undergo integral promotions before shifting)
     )
+    
+    $(P The expression evaluates according to
+
+        --------------
+        void doOpAssignment(ref int x, int y)
+        {
+            x = x op y;
```

Comment:
There is a cast missing in there. I did not put it in the forum thread as I used int explicitly, but it is needed in the general case. For instance :

  byte a = 3;
  a += 3; // This is ok because of the cast, but won't work because of int promotion without).